internal/provider/provider_test.go

```diff
@@ -151,6 +151,20 @@ func TestProviderConfigurex509InvalidFromEnv(t *testing.T) {
 	assert.Equal(t, "x509: certificate signed by unknown authority", err.Summary())
 }
 
+func TestProviderAllowsEmptyCACert(t *testing.T) {
```

Comment:
This test is failing for me locally:

```
 TF_ACC=1 TEST_CLOUD=lxd go test ./internal/provider/... -parallel 1  -run=TestProvider
--- FAIL: TestProviderAllowsEmptyCACert (29.01s)
    provider_test.go:164:
        	Error Trace:	/home/heather/work/src/github.com/juju/terraform-provider-juju/internal/provider/provider_test.go:164
        	Error:      	Not equal:
        	            	expected: "The ca_certificate provider property is not set and the Juju certificate authority is not trusted by your system"
        	            	actual  : "unable to connect to API: tls: failed to verify certificate: x509: cannot validate certificate for 10.41.184.184 because it doesn't contain any IP SANs"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-The ca_certificate provider property is not set and the Juju certificate authority is not trusted by your system
        	            	+unable to connect to API: tls: failed to verify certificate: x509: cannot validate certificate for 10.41.184.184 because it doesn't contain any IP SANs
        	Test:       	TestProviderAllowsEmptyCACert
    provider_test.go:165:
        	Error Trace:	/home/heather/work/src/github.com/juju/terraform-provider-juju/internal/provider/provider_test.go:165
        	Error:      	Not equal:
        	            	expected: "x509: certificate signed by unknown authority"
        	            	actual  : "Client Error"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-x509: certificate signed by unknown authority
        	            	+Client Error
        	Test:       	TestProviderAllowsEmptyCACert
FAIL
FAIL	github.com/juju/terraform-provider-juju/internal/provider	117.146s
FAIL
```

Reply:
That's interesting, running locally I'm also using Juju 3.5.3 and I've tried running the tests with -count=10 and it always passes. It seems like in the CI especially its flakey, sometimes passing.
I'm pushing up a fix but I don't fully understand why the error changes.
The expected error is (from the docstrings for the errors)
`UnknownAuthorityError` - results when the certificate issuer is unknown
`HostnameError` - results when the set of authorized names doesn't match the requested name.

Reply:
The test consistently failed for me when I ran it locally, not flakey at all.